hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodiePayloadConfig.java

```diff
@@ -107,7 +105,9 @@ public Builder withPayloadEventTimeField(String payloadEventTimeField) {
     }
 
     public HoodiePayloadConfig.Builder withPayloadClass(String payloadClassName) {
-      payloadConfig.setValue(PAYLOAD_CLASS_NAME, payloadClassName);
+      if (!StringUtils.isNullOrEmpty(payloadClassName)) {
+        payloadConfig.setValue(PAYLOAD_CLASS_NAME, payloadClassName);
+      }
```

Comment:
IMO it would be better to add a new `HoodieMergeConfig` class so that new merger code only uses `HoodieMergeConfig` instance and relevant configs (the class can still use the same config properties defined in this class).  We'll keep this class intact.  Once we get rid of the payload APIs, we will also remove `HoodiePayloadConfig` class.

Reply:
The `HoodieMergeConfig` should be a plain POJO containing the config values only.

Reply:
https://issues.apache.org/jira/browse/HUDI-8407